File: lista1/task0.py
import numpy as np



a = np.arange(1,101)
b = np.arange(1,100,2)
c = np.linspace(-1., 1, 201)*np.pi
d = np.append(np.linspace(-1, 0, 100, endpoint=False), np.linspace(0.01, 1, 100))*np.pi
e = np.maximum(np.sin(a), 0)

print(a, b, c, d, e, sep='\n')

A = a.reshape(10,10)
B = np.diag(a) + np.diag(a[:-1], k=1) + np.diag(a[:-1], k=-1)
C = np.triu(np.ones((10,10)))
D = np.append( np.cumsum(a), np.cumprod(a, dtype=np.float64) ).reshape(2,-1)
E = (np.mod(np.arange(0,100), np.arange(0,100).reshape(100,1)) == 0) * 1 # 0-indexed

print(A, B, C, D, E, sep='\n')

print(E[1 ,2], E[10,20], E[20,10])

File: list2/task0.m
a = [1:100];
b = [1:2:100];
c = [-1:0.01:1]*pi;
d = c;
d(101) = [];
e = max(sin(a),0);

A = reshape(a,[10,10]);
B = diag(a,0) + diag(a(99:-1:1),-1) + diag(a(99:-1:1),1);
C = triu(ones(10,10));
D = [cumsum(a);cumprod(a)];
E = bsxfun(@mod, a, a') == 0;
